ext/MitsubaLoader/CMitsubaLoader.cpp

```diff
@@ -13,6 +13,151 @@ namespace ext
+		//TODO int_ext_ior
+		break;
+	case CElementBSDF::Type::COATING:
+		_IRR_FALLTHROUGH;
+	case CElementBSDF::Type::ROUGHCOATING:
+		bitfields |= _node->coating.distribution << bsdf::BITFIELDS_SHIFT_NDF;
+		bitfields |= static_cast<uint32_t>(_node->coating.alpha.value.type==SPropertyElementData::INVALID)<<bsdf::BITFIELDS_SHIFT_ALPHA_U_TEX;
+		if (_node->coating.distribution == CElementBSDF::RoughSpecularBase::ASHIKHMIN_SHIRLEY)
+			bitfields |= static_cast<uint32_t>(_node->coating.alphaV.value.type==SPropertyElementData::INVALID)<<bsdf::BITFIELDS_SHIFT_ALPHA_V_TEX;
+		bitfields |= static_cast<uint32_t>(_node->coating.sigmaA.value.type==SPropertyElementData::INVALID)>>bsdf::BITFIELDS_SHIFT_SIGMA_A_TEX;
+		if (_node->coating.alpha.value.type==SPropertyElementData::FLOAT)
+			reinterpret_cast<float&>(retval.coating.alpha_u[0]) = _node->coating.alpha.value.fvalue;
+		if (_node->coating.distribution==CElementBSDF::RoughSpecularBase::ASHIKHMIN_SHIRLEY && _node->coating.alphaV.value.type==SPropertyElementData::FLOAT)
+			reinterpret_cast<float&>(retval.coating.alpha_v[0]) = _node->coating.alphaV.value.fvalue;
+		retval.coating.thickness = _node->coating.thickness;
+		if (_node->coating.sigmaA.value.type==SPropertyElementData::SPECTRUM)
+			retval.coating.sigmaA = rgb32f_to_rgb19e7(reinterpret_cast<const uint32_t*>(_node->coating.sigmaA.value.vvalue.pointer));
+		break;
+	case CElementBSDF::Type::BUMPMAP:
+		break;
+	case CElementBSDF::Type::PHONG:
+		assert(0);
+		break;
+	case CElementBSDF::Type::WARD:
+		bitfields |= _node->ward.variant << bsdf::BITFIELDS_SHIFT_WARD_VARIANT;
+		bitfields |= static_cast<uint32_t>(_node->ward.alphaU.value.type==SPropertyElementData::INVALID)<<bsdf::BITFIELDS_SHIFT_ALPHA_U_TEX;
+		bitfields |= static_cast<uint32_t>(_node->ward.alphaV.value.type==SPropertyElementData::INVALID)<<bsdf::BITFIELDS_SHIFT_ALPHA_V_TEX;
+		if (_node->ward.alphaU.value.type==SPropertyElementData::FLOAT)
+			reinterpret_cast<float&>(retval.ward.alpha_u[0]) = _node->ward.alphaU.value.fvalue;
+		if (_node->ward.alphaV.value.type==SPropertyElementData::FLOAT)
+			reinterpret_cast<float&>(retval.ward.alpha_v[0]) = _node->ward.alphaV.value.fvalue;
+		break;
+	case CElementBSDF::Type::MIXTURE_BSDF:
+		//TODO mixture will be translated into multiple blend BSDFs
+		break;
+	case CElementBSDF::Type::BLEND_BSDF:
+		bitfields |= static_cast<uint32_t>(_node->blendbsdf.weight.value.type==SPropertyElementData::INVALID)<<bsdf::BITFIELDS_SHIFT_WEIGHT_TEX;
+		reinterpret_cast<float&>(retval.blend.weight[0]) = _node->blendbsdf.weight.value.fvalue;
+		break;
+	case CElementBSDF::Type::MASK:
+		bitfields |= static_cast<uint32_t>(_node->mask.opacity.value.type==SPropertyElementData::INVALID)<<bsdf::BITFIELDS_SHIFT_OPACITY_TEX;
+		if (_node->mask.opacity.value.type==SPropertyElementData::SPECTRUM)
+			retval.mask.opacity = rgb32f_to_rgb19e7(reinterpret_cast<const uint32_t*>(_node->mask.opacity.value.vvalue.pointer));
+		break;
+	case CElementBSDF::Type::TWO_SIDED:
+
```

Comment:
bitfield

Reply:
still an issue

Reply:
this code is not there any more and twosided already is a bitfield